src/strategy/redis_cache_provider.ts

```diff
@@ -22,154 +22,153 @@ const SAMLRequestCacheItem = t.interface({
 });
 
 export interface IExtendedCacheProvider {
-  save: (
+  readonly save: (
     RequestXML: string,
     samlConfig: SamlConfig
   ) => TaskEither<Error, SAMLRequestCacheItem>;
-  get: (AuthnRequestID: string) => TaskEither<Error, SAMLRequestCacheItem>;
-  remove: (AuthnRequestID: string) => TaskEither<Error, string>;
+  readonly get: (
+    AuthnRequestID: string
+  ) => TaskEither<Error, SAMLRequestCacheItem>;
+  readonly remove: (AuthnRequestID: string) => TaskEither<Error, string>;
 }
 
 // those methods must never fail since there's
 // practically no error handling in passport-saml
 // (a very bad lot of spaghetti code)
-export const noopCacheProvider = (): CacheProvider => {
-  return {
-    // saves the key with the optional value
-    // invokes the callback with the value saved
-    save(_, value, callback): void {
-      const v = {
-        createdAt: new Date(),
-        value
-      };
-      callback(null, v);
-    },
-    // invokes 'callback' and passes the value if found, null otherwise
-    get(_, callback): void {
-      callback(null, {});
-    },
-    // removes the key from the cache, invokes `callback` with the
-    // key removed, null if no key is removed
-    remove(key, callback): void {
-      callback(null, key);
-    }
-  };
-};
+export const noopCacheProvider = (): CacheProvider => ({
+  // saves the key with the optional value
+  // invokes the callback with the value saved
+  // eslint-disable-next-line prefer-arrow/prefer-arrow-functions
+  save(_, value, callback): void {
```

Comment:
We can use arrow function even here like all the changes before to avoid the eslint disable comments.
```suggestion
  save: (_, value, callback): void => {
```